ДЗ20 Клиент-серверное приложение "Калькулятор"

EvgeniiKovalev opened the pull request.

EvgeniiKovalev committed:
ДЗ20 старт

EvgeniiKovalev committed:
ДЗ20 готово,
каждая коммуникация клиента и сервера (запрос-ответ) завершается отключение клиента от сервера. Сервер при этом не завершает работу. В цикле новая коммуникация клиента начинается с подключения к серверу

EvgeniiKovalev committed:
ДЗ20
каждая коммуникация клиента и сервера  (запрос-ответ) теперь выполняется в рамках одного подключения, т.е.  коммуникация не завершается закрытием соединения м/у сервером и клиентом.

Сервер научился обрабатывать потерю соединения с клиентом и не падает.

EvgeniiKovalev committed:
ДЗ20 fix

EvgeniiKovalev committed:
ДЗ20 fix

EvgeniiKovalev committed:
Merge remote-tracking branch 'origin/Feature/homework20' into Feature/homework20

EvgeniiKovalev committed:
Доработал shutdown server
и вывод ip и порта клиента на стороне сервера

EvgeniiKovalev committed:
ДЗ20 добавил пакет exercisemachine в котором решены 5 задач тренажера, которые шли к этому дз

a-poliakov commented on src/main/java/ru/otus/java/basic/homeworks/homework20/exercisemachine/ExceptionTaskMain.java:
```diff
@@ -0,0 +1,38 @@
+        try {
+            callable.call();
+        } catch (Exception e) {
+            return e.getClass() == clazz;
```
нельзя сравнивать на равенство два объекта, e.getClass() == clazz;

a-poliakov commented on src/main/java/ru/otus/java/basic/homeworks/homework20/exercisemachine/ExceptionTaskBuggyMethod.java:
```diff
@@ -0,0 +1,28 @@
+        int result = 0;
+        for (var item: inputNumbers) {
+            try {
+                result += (item + 1) / item;
```
здесь обратите внимание на условие: "а результат вычисления итерации, на которой была получена ошибка, был бы равен 0"

EvgeniiKovalev committed:
ДЗ20
1.добавил вывод списка мат. операций на клиенте, реализовал по запросу клиента. При подключении клиент автоматически без участия пользователя запрашивает доступные мат. операции.

2. переписал сервер с цель инкапсулировать подключение клиента к серверу, чтобы можно было реализовать обслуживание нескольких одновременных подключений от одного клиентского ip
